packages/api/src/config.js

```diff
@@ -0,0 +1,321 @@
+    },
+
+    maintenanceMode: maintenanceModeFromString(configVars.MAINTENANCE_MODE),
+
+    secrets: {
+      salt: configVars.SALT,
+      metaplexAuthToken: configVars.METAPLEX_AUTH_TOKEN,
+      ucanPrivateKey: configVars.PRIVATE_KEY,
+    },
+
+    external: {
+      cluster: {
+        url: clusterUrl,
+        basicAuthToken: configVars.CLUSTER_BASIC_AUTH_TOKEN,
+      },
+
+      database: {
+        url: configVars.DATABASE_URL,
+        authToken: configVars.DATABASE_TOKEN,
+      },
+
+      s3: {
+        endpoint: configVars.S3_ENDPOINT,
+        region: configVars.S3_REGION,
+        accessKeyId: configVars.S3_ACCESS_KEY_ID,
+        secretAccessKey: configVars.S3_SECRET_ACCESS_KEY,
+        bucketName: configVars.S3_BUCKET_NAME,
+      },
+
+      magicLink: {
+        secret: configVars.MAGIC_SECRET_KEY,
+      },
+
+      logtail: {
+        authToken: configVars.LOGTAIL_TOKEN,
+      },
+
+      sentry: {
+        dsn: configVars.SENTRY_DSN,
+      },
+
+      mailchimp: {
+        apiKey: configVars.MAILCHIMP_API_KEY,
+      },
+    },
+  }
+}
+
+/**
+ * Loads configuration variables from the global environment and returns a JS object
+ * keyed by variable names.
+ *
+ * Exported for testing. See {@link getServiceConfig} for main config accessor.
+ *
+ * @returns { Record<string, string>} an object with `vars` containing all found variables and their values, and a `notFound` array of variables that were not defined
+ */
+export function loadConfigVariables() {
+  /** @type Record<string, string> */
+  const vars = {}
+
+  /** @type Record<string, unknown> */
+  const globals = globalThis
+
+  for (const name of Object.keys(DEFAULT_CONFIG_VALUES)) {
+    const val = globals[name]
+    if (typeof val === 'string') {
+      vars[name] = val
+    }
+  }
+  return vars
+}
+
+/**
+ * Merges `vars` with default values if any keys are missing, iff default values
+ * are allowed in the current runtime env (see {@link allowDefaultConfigValues}).
+ *
+ * In "strict" environments (staging & production), this function will throw
+ * if `vars` does not contain a value for each key in {@link DEFAULT_CONFIG_VALUES}.
+ *
+ * In "lenient" environments (test & dev), this function will mutate `vars` in-place
+ * and set any missing keys to the value from {@link DEFAULT_CONFIG_VALUES}.
+ *
+ * @param {Record<string, string>} vars
+ */
+function mergeDefaultValuesIfAllowed(vars) {
+  // in "strict" environments (staging & production), we force all config variables
+  // to have a defined value. In dev & test, we allow falling back to default config values.
+  const env = runtimeEnvFromString(vars['ENV'])
+  if (allowDefaultConfigValues(env)) {
+    mergeWithDefaultValues(vars, env)
+
+    // special case for VERSION, because ucan-storage defines a VERSION global
+    // that seems to be clobbering ours if no VERSION var is defined by cloudflare.
+    if (vars['VERSION'] === '0.8.0') {
+      vars['VERSION'] = DEFAULT_CONFIG_VALUES.VERSION
+    }
```

Comment:
this happens because of this https://github.com/nftstorage/nft.storage/blob/main/packages/api/scripts/cli.js#L61-L63 we should really namespace the those `NFT_STORAGE_VERSION` etc

Reply:
That's what I thought too, but that would set version to a more complex string: `${pkg.name}@${pkg.version}-${opts.env}+${git.short(__dirname)}`

Here's what shows up in `dist.js`:

![image](https://user-images.githubusercontent.com/678715/169597974-034a53b7-6dd6-4d11-9bc7-730b2afc851b.png)

Also, if I manually apply the edit from [here](https://github.com/nftstorage/ucan.storage/pull/28) to the copy `ucan-storage` inside of `node_modules` (or yarn link that branch), the version test passes again.